cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -41,41 +48,52 @@ def time(self):
-        'daint:mc': {'time': (138.0, None, 0.15, 's')},
-        'eiger:mc': {'time': (100.0, None, 0.10, 's')},
-        'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
-    }
+    valid_systems = ['daint:mc', 'eiger:mc', 'pilatus:mc']
+    use_multithreading = False
 
     @run_after('init')
     def setup_by_scale(self):
-        self.descr = f'VASP CPU check'
-        if self.current_system.name == 'dom':
-            self.num_tasks = 72
-            self.num_tasks_per_node = 12
-            self.use_multithreading = True
-        elif self.current_system.name in ['eiger', 'pilatus']:
-            self.num_tasks = 64
-            self.num_tasks_per_node = 4
-            self.num_cpus_per_task = 8
-            self.num_tasks_per_core = 1
-            self.use_multithreading = False
-            self.variables = {
-                'MPICH_OFI_STARTUP_CONNECT': '1',
-                'OMP_NUM_THREADS': str(self.num_cpus_per_task),
-                'OMP_PLACES': 'cores',
-                'OMP_PROC_BIND': 'close'
-            }
+        self.descr = (f'VASP CPU check (version: {self.scale}, '
+                      f'{self.variant})')
+        if self.scale == 'small':
+            self.valid_systems += ['dom:mc']
+            if self.current_system.name in ['daint', 'dom']:
+                self.num_tasks = 12
+                self.num_tasks_per_node = 2
+                self.num_cpus_per_task = 18
+            elif self.current_system.name in ['eiger', 'pilatus']:
+                self.num_tasks = 64
+                self.num_tasks_per_node = 4
+                self.num_cpus_per_task = 8
+                self.num_tasks_per_core = 1
+                self.variables = {
+                    'MPICH_OFI_STARTUP_CONNECT': '1',
+                    'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+                    'OMP_PLACES': 'cores',
+                    'OMP_PROC_BIND': 'close'
+                }
         else:
-            self.num_tasks = 32
-            self.num_tasks_per_node = 2
-            self.use_multithreading = True
-
-        self.reference = self.references_by_variant[self.variant]
-        self.tags |= {'maintenance', 'production'}
+            if self.current_system.name in ['daint']:
+                self.num_tasks = 32
+                self.num_tasks_per_node = 2
+                self.num_cpus_per_task = 18
+
+    @run_before('performance')
+    def set_reference(self):
+        references = {
+            'small': {
+                'dom:mc': {'time': (65.0, None, 0.15, 's')},
+                'daint:mc': {'time': (65.0, None, 0.15, 's')},
+                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
+                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
+            },
+            'large': {
+                'daint:mc': {'time': (55.0, None, 0.15, 's')},
+                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
+                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
+            }
+        }
```

Comment:
I prefer to define the references in a dictionary in the class body. The reason is that you have the significant information of the test right at the beginning.

Reply:
I defined the references in the class body at the beginning (using the auto-detected architectures as well)